### PR TITLE
Add paegan-transport

### PR DESCRIPTION
**Pull request by daf**

### CODE REVIEW

File: paegan-transport/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        - python
+        - setuptools
+    run:
+        - python
+        - paegan
+        - fiona
+        - requests
+        - redis
+        - geojson
+        - pytables
```
Comment (ocefpaf) [inline paegan-transport/meta.yaml]: @daf apparently you are missing `redis` here.  (It is available in conda.)

Comment (daf) [inline paegan-transport/meta.yaml]: note line 22...

Comment (daf) [inline paegan-transport/meta.yaml]: @ocefpaf i'm guessing a channel issue?

Comment (ocefpaf) [inline paegan-transport/meta.yaml]: It is available in the default channel.  Let me check it here...

File: paegan-transport/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+requirements:
+    build:
+        - python
+        - setuptools
+    run:
+        - python
+        - paegan
+        - fiona
+        - requests
+        - redis
```
Comment (ocefpaf) [inline paegan-transport/meta.yaml]: Change to `redis-py`

**Comment by daf** [comment]
@ocefpaf using `redis-py` fixed travis and broke appveyor :(

**Comment by ocefpaf** [comment]
A quick search binstar search does not show `redis-py` for Windows :unamused:

I am unsure is [this one](https://binstar.org/Trentonoliphant/redis/files) is the right redis due to the naming issue.  If redis is an optional dependency we can add a `# [not win]` in the `meta.yaml` file, but that will require some changes in the `paegan-transport`.

PS: Is redis a pure python module?  What are the dependencies?  How difficult would it be to create a Windows recipe?

**Comment by daf** [comment]
If i'm reading this properly, it has no deps at all: https://github.com/andymccurdy/redis-py/blob/master/setup.py

I've no experience creating a Windows recipe but could try it.

**Comment by ocefpaf** [comment]
> I've no experience creating a Windows recipe but could try it.

Just do the same thing as you did with pysgrid.  (Note that as long as you have a `bld.bat` you have a windows recipe.)

**Comment by ocefpaf** [comment]
BTW, the README says:

> redis-py requires a running Redis server. See Redis's quickstart for installation instructions.

Could that server be the Windows roadblock?

**Comment by daf** [comment]
I think it just means the library won't work without a running redis server to connect to (seems pretty self-evident to me).

**Comment by ocefpaf** [comment]
- Is `redis-py` (#302) without [the redis server](http://redis.io/) enough for `paegan-transport` to work properly?

- Can `redis-py` be an optional dependency of  `paegan-transport`?

**Comment by daf** [comment]
- Yes
- Also yes (should I change the recipe to indicate this, or change paegan-transport to late import redis stuff, or both?)

**Comment by ocefpaf** [comment]
> Also yes (should I change the recipe to indicate this, or change paegan-transport to late import redis stuff, or both?

Both!  That is a much better solution then providing a windows package that might be broken.

**Comment by daf** [comment]
Agreed 100%.

**Comment by daf** [comment]
How does one specify an optional deps in the conda recipe?

**Comment by ocefpaf** [comment]
`# [not win]` in front on it.

**Comment by ocefpaf** [comment]
See more at http://conda.pydata.org/docs/building/meta-yaml.html#preprocessing-selectors

**Comment by daf** [comment]
@ocefpaf good now?

File: paegan-transport/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package:
+    name: paegan-transport
+    version: "1.0.3dev"
+
+source:
+  git_url: https://github.com/axiom-data-science/paegan-transport.git
+  git_tag: db8f957f5f379def417f93c1941d3f20dbbf16bb
```
Comment (ocefpaf) [inline paegan-transport/meta.yaml]: Loved it!  Avoid master as much as you can.  (Releases as better though :stuck_out_tongue_winking_eye:)

File: paegan-transport/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        - requests
+        - redis-py  # [not win]
+        - geojson
+        - pytables
+
+test:
+    imports:
+        - paegan.transport
+        - paegan.transport.bathymetry
+        - paegan.transport.controllers # [not win]
```
Comment (ocefpaf) [inline paegan-transport/meta.yaml]: Just out of curiosity.  This is due to the lack of `redis-py` or something else?

**Comment by ocefpaf** [comment]
> @ocefpaf good now?

It was good before.  It is perfect now!  Thanks!!

**Pull request merged**